docs_sphinx/advanced/custom_events.rst

```diff
@@ -1,21 +1,78 @@
 Custom events
 =============
 
+Overview
+--------
+
 In most simulations, a `NeuronGroup` defines a threshold on its membrane
 potential that triggers a spike event. This event can be monitored by a
 `SpikeMonitor`, it is used in synaptic interactions, and in integrate-and-fire
 models it also leads to the execution of one or more reset statements.
 
 Sometimes, it can be useful to define additional events, e.g. when an ion
-concentration in the cell crosses a certain threshold. This can be done with
+concentration in the cell crosses a certain threshold. This can be done
+with the custom events system in Brian, which is illustrated in this diagram.
+
+.. image:: custom_events.svg
+
+You can see in this diagram that the source `NeuronGroup` has four types
+of events, called ``spike``, ``evt_other``, ``evt_mon`` and ``evt_run``.
```

Comment:
I like the example and the figure, I just wonder whether using three separate custom events might make some users think that you *need* to have separate events when you want to do e.g. both monitoring and propagation? Of course the `spike` event should make clear that this is not the case, but maybe a sentence along the lines of "we use separate events here for clarity, but the same event could be used for everything"?